packages/squiggle-lang/src/rescript/Utility/E.res

```diff
@@ -289,8 +289,14 @@ module A = {
       ))
       |> Rationale.Result.return
     }
-  let rangeFloat = (~step=1, start, stop) =>
-    Belt.Array.rangeBy(start, stop, ~step) |> fmap(Belt.Int.toFloat)
+
+  let diff = (arr: array<float>): array<float> =>
```

Comment:
I think I prefer `zipWithMinus`, btw. 

you wouldn't call `zipWithPlus` a `sum` now would you?

Reply:
Are you joking about sum xD.

anything with "zip" in it infers too lists. If I was to rename this, I might call it `differenceList`. But my Haskell intuition says that this function is called `diff`.

These guys seem to agree with me:
https://stackoverflow.com/questions/9512513/how-do-you-compute-the-difference-between-successive-elements-of-a-list-of-unkno

Reply:
(i've personally enjoyed the word `zip` independendent of if I'm talking about lists, arrays, or vectors) but ok, `diff` isn't the end of the world for me I juts think it reads _slightly_ awkwardly

Reply:
At very least leave a comment above it describing what it's doing, maybe with a link to that. ``diff`` isn't clear to me either.

Reply:
Also, this should go in A.Floats, which is for arrays of floats.